examples/applications/object_detection/retina_net/basic/pascal_voc/loader.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import matplotlib.pyplot as plt
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+from keras_cv import bounding_box
+
+IMG_SIZE = (256, 256)
+BATCH_SIZE = 9
+
+
+def unpackage(format):
+    """mapping function to create batched image and bbox coordinates"""
+
+    def apply(inputs):
+        inputs["image"] = tf.image.resize(inputs["image"], IMG_SIZE)
+        inputs["objects"]["bbox"] = bounding_box.convert_format(
+            inputs["objects"]["bbox"],
+            images=inputs["image"],
+            source="rel_yxyx",
+            target=format,
+        )
+
+        bounding_boxes = inputs["objects"]["bbox"]
+        labels = tf.cast(inputs["objects"]["label"], tf.float32)
+        labels = tf.expand_dims(labels, axis=-1)
+        bounding_boxes = tf.concat([bounding_boxes, labels], axis=-1)
+        return {"images": inputs["image"], "bounding_boxes": bounding_boxes}
+
+    return apply
+
+
+def load_pascal_voc(split, bounding_box_format, batch_size):
+    dataset, dataset_info = tfds.load(
+        "voc/2007", split=split, shuffle_files=True, with_info=True
+    )
+    dataset = dataset.map(
+        unpackage(format=bounding_box_format), num_parallel_calls=tf.data.AUTOTUNE
+    )
+    dataset = dataset.apply(
+        tf.data.experimental.dense_to_ragged_batch(batch_size=batch_size)
+    )
+    return dataset, dataset_info
+
+
+def main():
```

Comment:
Usually the main/demo should live outside of the lib.

Reply:
This script doesn't live in the main lib, the loader is part of the example in this case.

I'd like to wrap the loader into the library in a follow up PR.